### PR TITLE
FEATURE: QA-20139 write auto for Table Transpose - Insight View and Table Component

### PR DESCRIPTION
**Pull request by maitrieuphilong**
JIRA: QA-20139

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16044/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17530/ : SUCCESS in 7m 30s

File: libs/sdk-ui-tests-e2e/cypress/integration/04-sdk-ui-insight/tableTranspose.spec.ts
Comment (maitrieuphilong) [inline libs/sdk-ui-tests-e2e/cypress/integration/04-sdk-ui-insight/tableTranspose.spec.ts:1]: draft file, will delete it on next commit

File: libs/sdk-ui-tests-e2e/cypress/integration/03-sdk-ui-charts/pivotTableTransposed.spec.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+// (C) 2023 GoodData Corporation
+import * as Navigation from "../../tools/navigation";
+import { Table } from "../../tools/table";
+
+describe("Table Component", { tags: ["checklist_integrated_tiger"] }, () => {
+    const table = new Table(".s-table-component-transpose");
+
+    it("should display Metric in row, Column header on top", () => {
+        Navigation.visit("visualizations/pivot-table/pivot-table-transposed-has-mr-row-left");
+        table.waitLoaded().hasMetricHeaderInRow(1, 1, "Amount").hasMetricHeaderInRow(2, 1, "Amount");
```
Comment (thao-luong) [inline libs/sdk-ui-tests-e2e/cypress/integration/03-sdk-ui-charts/pivotTableTransposed.spec.ts:10]: Missing check column header on Top

File: libs/sdk-ui-tests-e2e/cypress/integration/03-sdk-ui-charts/pivotTableTransposed.spec.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+describe("Insight View", { tags: ["checklist_integrated_tiger"] }, () => {
+    const table = new Table(".s-insight-view-transpose");
+
+    it("should display Metric in row, Column header on left", () => {
+        Navigation.visit("insight/insight-transpose-has-mc-row-left");
+        table.waitLoaded().hasCellValue(0, 0, "Forecast Category").hasMetricHeaderInRow(1, 0, "Amount");
+    });
+
+    it("should display Metric in column, Column header on top", () => {
+        Navigation.visit("insight/insight-transpose-has-mc-column-top");
```
Comment (myduong243) [inline libs/sdk-ui-tests-e2e/cypress/integration/03-sdk-ui-charts/pivotTableTransposed.spec.ts:48]: The insightView "insight-transpose-has-mc-column-top", but the config is "column - lefts", so please rename this to column - left

File: libs/sdk-ui-tests-e2e/cypress/integration/03-sdk-ui-charts/pivotTableTransposed.spec.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            .hasMeasureHeader(0, "Amount", true)
+            .hasMeasureHeader(1, "Amount", true);
+    });
+
+    it("should display Metric in row", () => {
+        Navigation.visit("insight/insight-transpose-has-mc-row");
+        table.waitLoaded().hasMetricHeaderInRow(0, 0, "Amount");
+    });
+
+    it("should display Column header on top", () => {
```
Comment (myduong243) [inline libs/sdk-ui-tests-e2e/cypress/integration/03-sdk-ui-charts/pivotTableTransposed.spec.ts:70]: Wrong case in E2E, this case should config with 
`config={     
            columnHeadersPosition:”left”
 }`
Please help me to update it, many thanks!

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16055/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17541/ : FAILURE in 7m 18s

File: libs/sdk-ui-tests-e2e/cypress/integration/03-sdk-ui-charts/pivotTableTransposed.spec.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
+// (C) 2023 GoodData Corporation
+import * as Navigation from "../../tools/navigation";
+import { Table } from "../../tools/table";
+
+describe("Table Component", { tags: ["checklist_integrated_tiger"] }, () => {
+    const table = new Table(".s-table-component-transpose");
+
+    it("should display Metric in row, Column header on top", () => {
+        Navigation.visit("visualizations/pivot-table/pivot-table-transposed-has-mr-row-left");
+        table.waitLoaded().hasMetricHeaderInRow(1, 1, "Amount").hasMetricHeaderInRow(2, 1, "Amount").hasHeader("Product");;
```
Comment (myduong243) [inline libs/sdk-ui-tests-e2e/cypress/integration/03-sdk-ui-charts/pivotTableTransposed.spec.ts:10]: Redundant ; in this line

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16061/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17545/ : FAILURE in 15m 01s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16068/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17549/ : SUCCESS in 7m 13s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16069/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17550/ : FAILURE in 7m 03s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16070/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17552/ : SUCCESS in 7m 12s

**Comment by maitrieuphilong** [comment]
run passed on my branch:
https://checklist.intgdc.com/job/cypress-tests/job/gooddata-ui-sdk-cypress-checklist-worker/1180/console

**Comment by myduong243** [comment]
LGTM

**Review by myduong243 — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4163

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3280/ : SUCCESS in 10m 37s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17553/ : SUCCESS in 7m 18s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/895/ : SUCCESS in 10m 06s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4440/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1160/ : FAILURE in 1m 30s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/800/ : SUCCESS in 9m 33s

**Pull request merged**